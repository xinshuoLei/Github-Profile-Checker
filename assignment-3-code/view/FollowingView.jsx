import * as React from 'react';
import { useState, useEffect } from 'react';
import {
   Text, View, ScrollView, FlatList, StyleSheet, Dimensions, Image, Button, TouchableOpacity
} from 'react-native';
import ApiRequest from '../model/APIRequest';
import FollowerAndFollowing from '../model/FollowerAndFollowingModel';
 
 
/**
* The screen for following
*/
const FollowingView = ({navigation}) => {
  const [isLoading, setLoading] = useState(true);
  const [data, setData] = useState(null);

  // call api request to get data needed
  useEffect(() => {
    ApiRequest("IjzerenHein")
      .then((json) => setData(new FollowerAndFollowing(json)))
      .finally(() => setLoading(false));
  }, []);


  const renderItem = ({item}) => {
    return (
      <View style = {followerStyle.repoView}>
        <Image source={{ uri: item.avatarUrl }} style={followerStyle.avatar} />
        <Text style = {{fontSize: 20, marginLeft: 15, fontWeight: 'bold'}}>{item.login}</Text>
        <TouchableOpacity
          onPress={() => navigation.navigate('User Profile',  {user: item.login,})}>
          <Image source={require('../assets/link.png')} style = {{width: 15, height : 15, marginLeft : 5, tintColor: '#0645AD'}}/>
        </TouchableOpacity>
        <Text style = {{fontSize: 17, marginLeft: 15}}>{item.name ? item.name : 'N/A'}</Text>
      </View>
    )
  }

  return (
    <ScrollView>
      {isLoading ? (
        // display a text indicating view is loading if loading not completed
        (<Text>Loading</Text>)
      ) : (
      <View style = {{justifyContent: 'center', alignItems: 'center'}}>
        <FlatList
          contentContainerStyle={{justifyContent: 'center'}}
          data = {data.following}
          renderItem = {renderItem}>
      
        </FlatList>
      </View>)}
    </ScrollView>
  );
};

/**
* style for components in profile view
*/
const followerStyle = StyleSheet.create({
  repoView: {
    flexDirection : 'row', 
    height: 75, 
    width : Dimensions.get('window').width,
    backgroundColor : '#D9DBDA',
    marginTop: 10,
    alignItems: 'center'
  },
  avatar : {
    width: 50, 
    height: 50, 
    marginLeft: 35, 
    borderRadius: 50 / 2
  }
});

export default FollowingView;
