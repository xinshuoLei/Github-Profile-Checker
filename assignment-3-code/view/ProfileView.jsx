import * as React from 'react';
import { useState, useEffect } from 'react';
import {
  Text, View, SafeAreaView, Image, StyleSheet, Button,
} from 'react-native';
import ApiRequest from '../model/APIRequest';
import ProfileModel from '../model/ProfileModel';

/**
 * style for components in profile view
 */
const profileStyle = StyleSheet.create({
  largeText: {
    fontSize: 20,
    fontWeight: 'bold',
  },
  firstText: {
    fontSize: 15,
    marginTop: 10,
  },
  regularText: {
    fontSize: 15,
    marginTop: 8,
  },
});

/**
 * screen for user profile
 */
const ProfileView = ({ route, navigation }) => {
  const { user } = route.params;
  const [isLoading, setLoading] = useState(true);
  const [data, setData] = useState(null);

  // call api request to get data needed
  useEffect(() => {
    ApiRequest(user)
      .then((json) => setData(new ProfileModel(json)))
      .finally(() => setLoading(false));
  }, [user]);

  return (
    <SafeAreaView style={{ justifyContent: 'center', alignItems: 'center' }}>
      {isLoading ? (
        // display a text indicating view is loading if loading not completed
        (<Text>Loading</Text>)
      )
      // display profile when loading finishes
        : (
          <View style={{ flexDirection: 'column' }}>
            <View style={{ flex: 1, flexDirection: 'row', justifyContent: 'center' }}>
              <View style={{ alignItems: 'center', marginTop: 30 }}>
                <Image source={{ uri: data.avatar }} style={{ width: 150, height: 150 }} />
              </View>
              <View style={{ marginLeft: 30, marginTop: 30 }}>
                <Text style={profileStyle.largeText}>{data.userName ? data.userName : 'no username found'}</Text>
                <Text style={profileStyle.firstText}>{data.name ? data.name : 'no name found'}</Text>
                <Text style={profileStyle.regularText}>{data.email ? data.email : 'no email found'}</Text>
                <Text style={profileStyle.regularText}>{data.website ? data.website : 'no website found'}</Text>
                <Text />
              </View>
            </View>
            <View style={{
              width: 400, marginTop: 50, justifyContent: 'center',
            }}
            >
              <Text style={profileStyle.regularText}>
                created at:
                {data.createdAt}
              </Text>
              <Text style={profileStyle.regularText}>{data.bio}</Text>
              <Text style={profileStyle.regularText}>
                total number of followers:
                {data.followerCount}
              </Text>
              <Button
                title="Go to Folower"
                onPress={() => navigation.navigate('Navigator', { screen: 'Follower' })}
              />
              <Text style={profileStyle.regularText}>
                total number of followings:
                {data.followingCount}
              </Text>
              <Button
                title="Go to Following"
                onPress={() => navigation.navigate('Navigator', { screen: 'Following' })}
              />
              <Text style={profileStyle.regularText}>
                total number of public repositories:
                {data.repoCount}
              </Text>
              <Button
                title="Go to Repo"
                onPress={() => navigation.navigate('Navigator', { screen: 'Repo' })}
              />
            </View>
          </View>
        )}
    </SafeAreaView>
  );
};

export default ProfileView;
