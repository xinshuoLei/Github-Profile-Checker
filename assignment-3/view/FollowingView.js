import * as React from 'react';
import { Text, View } from 'react-native';

/**
 * The screen for following
 */
const FollowingView = ({navigation}) => {
  return (
    <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
      <Text>empty screen for following</Text>
    </View>
  );
}

export default FollowingView;

