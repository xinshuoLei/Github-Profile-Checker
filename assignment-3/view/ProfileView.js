import * as React from 'react';
import { useState, useEffect } from "react"
import { Text, View, SafeAreaView } from 'react-native';
import ApiRequest from '../model/APIRequest';

/**
 * screen for user profile
 */
const ProfileView = (props) => {
  const [isLoading, setLoading] = useState(true)
  const [data, setData] = useState([])

  useEffect(() => {
    ApiRequest()
    .then((json) => setData(json.data.user.websiteUrl))
    .finally(() => setLoading(false))
  }, [])

  return (
    <SafeAreaView style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
      {isLoading? (
        (<Text>Loading</Text>)
      ):
      (<Text>{data}</Text>
      )}
    </SafeAreaView>
  );
}

export default ProfileView;