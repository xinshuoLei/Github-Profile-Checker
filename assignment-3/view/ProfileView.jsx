import * as React from 'react';
import { useState, useEffect } from 'react';
import {
  Text, View, SafeAreaView, Image, StyleSheet,
} from 'react-native';
import ApiRequest from '../model/APIRequest';
import ProfileModel from '../model/model';

/**
 * screen for user profile
 */
const ProfileView = () => {
  const [isLoading, setLoading] = useState(true);
  const [data, setData] = useState(null);
  let test;

  // call api request to get data needed
  useEffect(() => {
    ApiRequest()
      .then((json) => setData(new ProfileModel(json)))
      .finally(() => setLoading(false));
  }, []);


  return (
    <SafeAreaView style={{ justifyContent: 'center' }}>
      {isLoading ? (
        // display a text indicating view is loading if loading not completed
        (<Text>Loading</Text>)
      )
      // display profile when loading finishes
        : (
          <View style={{ flexDirection: 'column' }}>
            <View style={{ flex: 1, flexDirection: 'row', justifyContent: 'center' }}>
              <View style={{ alignItems: 'center', marginTop: 30 }}>
                <Image source={{ uri: data.avatar }} style={{ width: 150, height: 150 }} />
              </View>
              <View style={{ marginLeft: 30, marginTop: 30 }}>
                <Text style={profileStyle.largeText}>{data.userName ? data.userName : 'no username found'}</Text>
                <Text style={profileStyle.firstText}>{data.name ? data.name : 'no name found'}</Text>
                <Text style={profileStyle.regularText}>{data.email ? data.email : 'no email found'}</Text>
                <Text style={profileStyle.regularText}>{data.website ? data.website : 'no website found'}</Text>
                <Text />
              </View>
            </View>
            <View style={{
              marginLeft: 30, width: 350, marginTop: 200, justifyContent: 'center',
            }}
            >
              <Text style={profileStyle.regularText}>
                created at:
                {data.createdAt}
              </Text>
              <Text style={profileStyle.regularText}>{data.bio}</Text>
            </View>
          </View>
        )}
    </SafeAreaView>
  );
};

/**
 * style for components in profile view
 */
const profileStyle = StyleSheet.create({
  largeText: {
    fontSize: 20,
    fontWeight: 'bold',
  },
  firstText: {
    fontSize: 15,
    marginTop: 10,
  },
  regularText: {
    fontSize: 15,
    marginTop: 5,
  },
});

export default ProfileView;
