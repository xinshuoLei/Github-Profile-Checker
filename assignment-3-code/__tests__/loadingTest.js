import React from 'react';
import renderer from 'react-test-renderer';
import ProfileView from '../view/ProfileView.jsx';
import FollowingView from '../view/FollowingView.jsx'
import FollowerView from '../view/FollwerView.jsx'


it('renders correctly - profile loading', () => {
  const tree = renderer.create(<FollowerView />).toJSON();
  expect(tree).toMatchSnapshot();
});


it('renders correctly - follower loading', () => {
  const tree = renderer.create(<ProfileView />).toJSON();
  expect(tree).toMatchSnapshot();
});


it('renders correctly - following loading', () => {
  const tree = renderer.create(<FollowingView />).toJSON();
  expect(tree).toMatchSnapshot();
});
