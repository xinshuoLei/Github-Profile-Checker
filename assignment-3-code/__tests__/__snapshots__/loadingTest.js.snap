// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly - follower loading 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Text>
    Loading
  </Text>
</RCTSafeAreaView>
`;

exports[`renders correctly - following loading 1`] = `
<RCTScrollView>
  <View>
    <Text>
      Loading
    </Text>
  </View>
</RCTScrollView>
`;

exports[`renders correctly - profile loading 1`] = `
<RCTScrollView>
  <View>
    <Text>
      Loading
    </Text>
  </View>
</RCTScrollView>
`;
